exoscale/provider.go

```diff
@@ -102,6 +102,13 @@ func Provider() terraform.ResourceProvider {
 				Description: fmt.Sprintf("Exoscale DNS API endpoint (by default: %s)", defaultDNSEndpoint),
 				DefaultFunc: schema.EnvDefaultFunc("EXOSCALE_DNS_ENDPOINT", defaultDNSEndpoint),
 			},
+			"environment": {
+				Type:     schema.TypeString,
+				Optional: true,
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{
+					"EXOSCALE_ENVIRONMENT",
```

Comment:
The eventual naming convention for API-related settings is `EXOSCALE_API_*`:

```suggestion
					"EXOSCALE_API_ENVIRONMENT",
```